cpp/json_ffi/conv_template.h

```diff
@@ -0,0 +1,61 @@
+#ifndef MLC_LLM_JSON_FFI_OPENAI_API_CONV_TEMPLATE_H
+#define MLC_LLM_JSON_FFI_OPENAI_API_CONV_TEMPLATE_H
+
+#include <iostream>
+#include <string>
+#include <vector>
+#include <map>
+#include <optional>
+#include <variant>
+#include <typeinfo>
+
+#include "../serve/data.h"
+#include "picojson.h"
+
+using namespace mlc::llm::serve;
+
+enum class MessagePlaceholders {
+    SYSTEM,
+    USER,
+    ASSISTANT,
+    TOOL,
+    FUNCTION
+};
+
+MessagePlaceholders message_placeholder_from_string(const std::string& role);
+
+namespace mlc{
+namespace llm{
+namespace json_ffi{
+struct Conversation {
```

Comment:
Thank you @Kartik14 and @shreygupta2809 for the push! Good to have some docstrings for this class and member functions for readability.